File: common/src/main/java/net/impleri/playerskills/integration/kubejs/numeric/NumericSkillJS.java
package net.impleri.playerskills.integration.kubejs.numeric;

import net.impleri.playerskills.integration.kubejs.skills.GenericSkillBuilderJS;
import net.impleri.playerskills.numeric.NumericSkill;
import net.minecraft.resources.ResourceLocation;

public class NumericSkillJS extends NumericSkill {
    public NumericSkillJS(Builder builder) {
        super(builder.id, builder.initialValue, builder.description);
    }

    public static class Builder extends GenericSkillBuilderJS<Double> {
        public Builder(ResourceLocation name) {
            super(name);
        }

        @Override
        public NumericSkillJS createObject() {
            return new NumericSkillJS(this);
        }
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/integrations/kubejs/skills/NumericSkillJS.kt
package net.impleri.playerskills.integrations.kubejs.skills

import net.impleri.playerskills.api.Skill
import net.impleri.playerskills.skills.numeric.NumericSkill
import net.minecraft.resources.ResourceLocation

class NumericSkillJS(builder: Builder) : NumericSkill(
  builder.id,
  builder.initialValue,
  builder.description,
  builder.options,
  builder.changesAllowed,
  builder.teamMode,
  builder.notify,
  builder.notifyKey
) {
  class Builder(name: ResourceLocation) : GenericSkillBuilderJS<Double>(name) {
    override fun createObject(): Skill<Double> {
      return NumericSkillJS(this)
    }
  }
}
